domain/application/service/application.go

```diff
@@ -145,6 +152,12 @@ type ApplicationState interface {
 	// unit doesn't exist.
 	GetApplicationIDByUnitName(ctx context.Context, name coreunit.Name) (coreapplication.ID, error)
 
+	// GetApplicationNameByUnitName returns the application name for the named unit.
+	//
+	// The following errors may be returned:
+	// - [uniterrors.UnitNotFound] if the unit does not exist
+	GetApplicationNameByUnitName(ctx context.Context, name coreunit.Name) (string, error)
```

Comment:
question: why not do  `coreunit.Name.Application()`.

Reply:
You're right, that was a good idea. But anyways I changed the state methods to take UUIDs so that's not needed anymore.